go.mod

```diff
@@ -48,7 +48,7 @@ require (
 	github.com/jbenet/goprocess v0.1.4
 	github.com/julienschmidt/httprouter v1.3.0
 	github.com/libp2p/go-doh-resolver v0.4.0
-	github.com/libp2p/go-libp2p v0.35.4
+	github.com/libp2p/go-libp2p v0.35.1-0.20240730155614-111d6df08497
```

Comment:
@MarcoPolo I got below `fx` error ([log](https://github.com/ipfs/kubo/actions/runs/10167054881/job/28118558695#step:5:8645)) in after switching to `111d6df08497` from https://github.com/libp2p/go-libp2p/pull/2889:

`failed to build host.Host: received non-nil error from function "github.com/ipfs/kubo/core/node/libp2p".Host (github.com/ipfs/kubo/core/node/libp2p/host.go:41): could not build arguments for function "reflect".makeFuncStub (reflect/asm_amd64.s:28): could not build value group transport.Transport[group="transport"]: missing dependencies for function "reflect".makeFuncStub (reflect/asm_amd64.s:28): missing type: libp2pwebrtc.ListenUDPFn`

Is this something obvious, or still wip?

Reply:
fwiw confirmed code from https://github.com/libp2p/go-libp2p/pull/2889 is used and loads fine. Can create objects that implement `libp2pwebrtc.ListenUDPFn`, and still get this error, so this looks like a bug specific to how `fx`  works. 

I got the same error when in Rainbow.

Reply:
My apologies, I forgot to add the provide in a certain code path for the autonat host. I just pushed a commit that fixes this.

Reply:
@MarcoPolo I'm OoO today, but checked the version from https://github.com/ipfs/kubo/pull/10463/commits/6513aedde13403c37d91d0756e00ee753f8c21e5 before going AFK, and seems and it seems to work fine, listener on the same port initialized correctly and I was able to run libp2p identify over `/udp/4001/quic-v1`, so no regression on existing transports, which is the most important.

I'll be back Thu but feel free to move forward with https://github.com/libp2p/go-libp2p/pull/2889.

(CI fails for unrelated reason (rhis PR downgraded boxo version)).